### PR TITLE
Upgrade to Rails 7

### PR DESCRIPTION
**Pull request by jbirdjavi**

### CODE REVIEW

File: .github/workflows/ruby.yml [hunk 1]
```diff
@@ -67,3 +64,6 @@ jobs:
 
       - name: StandardRB
         run: bundle exec standardrb --format simple
+
+      - name: JavaScript Audit
```
Comment (dbenton9) [inline .github/workflows/ruby.yml:68]: I'm guessing the `Javascript Audit` is replacing the `Install yarn dependencies`?

File: Dockerfile [hunk 2]
```diff
@@ -21,7 +21,6 @@ RUN apk --no-cache add \
     postgresql-libs \
     tzdata \
     nodejs \
-    yarn \
```
Comment (dbenton9) [inline Dockerfile:24]: Why is this getting dropped?

File: config/initializers/filter_parameter_logging.rb [hunk 3]
```diff
@@ -1,4 +1,8 @@
 # Be sure to restart your server when you modify this file.
 
-# Configure sensitive parameters which will be filtered from the log file.
-Rails.application.config.filter_parameters += %i[password]
+# Configure parameters to be filtered from the log file. Use this to limit dissemination of
+# sensitive information. See the ActiveSupport::ParameterFilter documentation for supported
+# notations and behaviors.
+Rails.application.config.filter_parameters += [
+  :passw, :secret, :token, :_key, :crypt, :salt, :certificate, :otp, :ssn
```
Comment (dbenton9) [inline config/initializers/filter_parameter_logging.rb:7]: do we not want to continue filtering `password` in addition to `passw`

**Review by dbenton9 — COMMENTED** [review]
That's a lot of tedious edits. I don't know how many of them Rails did auto-magically and how many you had to go in and do manually, but I'm grateful for ya tackling this.

File: .github/workflows/ruby.yml (hunk 1 above)
Comment (jbirdjavi) [reply .github/workflows/ruby.yml:68]: Not quite. `Install yarn dependencies` was basically just loading all the JavaScript packages into the `node_modules` directory. With Rails 7, we don't have to do that anymore. The `config/importmap.rb` file specifies all the JS files/packages we need and where to find them. This cuts out a LOT of JS dependencies compared to the previous approach.

`JavaScript Audit` here is just making sure that the versions of our JS packages don't have any security issues, which we've never done in this repository before.

File: Dockerfile (hunk 2 above)
Comment (jbirdjavi) [reply Dockerfile:24]: Related to the previous comment, no need for it anymore. :)

File: config/initializers/filter_parameter_logging.rb (hunk 3 above)
Comment (jbirdjavi) [reply config/initializers/filter_parameter_logging.rb:7]: These are actually regex matchers, so `passw` will also filter out `password` or `passwd` or anything like that.

File: .github/workflows/ruby.yml (hunk 1 above)
Comment (jbirdjavi) [reply .github/workflows/ruby.yml:68]: See https://github.com/rails/importmap-rails for more information on how the new way is set up.

**Review by dbenton9 — APPROVED** [review]

**Review by knutsenm — APPROVED** [review]

File: app/controllers/login_controller.rb [hunk 4]
```diff
@@ -3,7 +3,7 @@ def create
     authorize_params = OktaOauth.authorize_params(prompt: params[:prompt]&.to_s)
     session[:oauth_state] = authorize_params.state
     session[:return_to_url] = params[:return_to_url]
-    redirect_to authorize_params.redirect_url
+    redirect_to authorize_params.redirect_url, allow_other_host: true
```
[the remark this answers is not in the document]
Comment (jbirdjavi) [reply app/controllers/login_controller.rb:6]: So I learned recently that all the options in the new_framework_defaults file will be set if you bump the version in config/application.rb of the `config.load_defaults` setting. If you leave that setting, then you can turn on the new defaults one by one in the new_framework_defaults file, but I prefer not to have to do that if I don't have to.

**Pull request merged**